src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -198,8 +255,20 @@ protected override void OnKeyDown(KeyEventArgs e)
 
         protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
         {
-            _header = e.NameScope.Find<Control>("PART_Header");
+            if (_header is InputElement previousInputMethod)
+            {
+                previousInputMethod.DoubleTapped -= HeaderDoubleTapped;
+            }
+
+            var h = e.NameScope.Find<Control>("PART_Header");
```

Comment:
There doesn't seem to be any need for a new variable `h` here. I would just set directly to `_header` which is easier to read and less verbose.

Reply:
I can do that. I am just so used to that pattern for multithreaded programs when dealing with mutable properties.

Reply:
Well, in this case you are getting a reference to a control for event subscriptions and other internal usages. All of that code is sharing the same reference which doesn't change until the template is re-applied. So it is quite a bit different of a case than accessing a potentially mutable property.

If you were getting a property value out of the property system, yes, it's common to do that once at the start of the method. This is usually for performance though. There is only one UIThread in Avalonia right now -- although the new composition thread exists too. Anyway, you don't have to worry most of the time about state change in a single method in a control -- everything is running on the same UIThread.

Reply:
Addressed